subsys/power/device_pm.c

```diff
@@ -18,27 +18,19 @@ LOG_MODULE_DECLARE(power);
 #define DEVICE_PM_SYNC   BIT(0)
 #define DEVICE_PM_ASYNC  BIT(1)
 
-/* Device PM states */
-enum device_pm_state {
-	DEVICE_PM_STATE_ACTIVE = 1,
-	DEVICE_PM_STATE_SUSPENDED,
-	DEVICE_PM_STATE_SUSPENDING,
-	DEVICE_PM_STATE_RESUMING,
-};
+/*
+ * Lets just re-use device pm states for FSM but add these two
+ * transitioning states for internal usage.
+ */
+#define DEVICE_PM_RESUMING_STATE   (DEVICE_PM_OFF_STATE + 1U)
+#define DEVICE_PM_SUSPENDING_STATE (DEVICE_PM_RESUMING_STATE + 1U)
```

Comment:
I'd not define internal states based on the value of external ones, a change on the external side may create unwanted value clashes.

Reply:
I will unify this with that states in near future. But there was already a dependency between those states, but was very confuse having `DEVICE_PM_ACTIVE_STATE` and `DEVICE_PM_STATE_ACTIVE` :/